File: pkg/dev_compiler/test/codegen/expect/misc.js
dart_library.library('misc', null, /* Imports */[
  'dart_sdk'
], function(exports, dart_sdk) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const misc = Object.create(null);
  misc._Uninitialized = class _Uninitialized extends core.Object {
    _Uninitialized() {
    }
  };
  dart.setSignature(misc._Uninitialized, {
    constructors: () => ({_Uninitialized: [misc._Uninitialized, []]})
  });
  misc.UNINITIALIZED = dart.const(new misc._Uninitialized());
  misc.Generic$ = dart.generic(T => {
    class Generic extends core.Object {
      get type() {
        return dart.wrapType(misc.Generic);
      }
      m() {
        return core.print(dart.wrapType(T));
      }
    }
    dart.setSignature(Generic, {
      methods: () => ({m: [dart.dynamic, []]})
    });
    return Generic;
  });
  misc.Generic = misc.Generic$();
  misc.Base = class Base extends core.Object {
    Base() {
      this.x = 1;
      this.y = 2;
    }
    ['=='](obj) {
      return dart.is(obj, misc.Base) && obj.x == this.x && obj.y == this.y;
    }
  };
  misc.Derived = class Derived extends core.Object {
    Derived() {
      this.z = 3;
    }
    ['=='](obj) {
      return dart.is(obj, misc.Derived) && obj.z == this.z && super['=='](obj);
    }
  };
  misc._isWhitespace = function(ch) {
    return ch == ' ' || ch == '\n' || ch == '\r' || ch == '\t';
  };
  dart.fn(misc._isWhitespace, core.bool, [core.String]);
  misc.expr = 'foo';
  misc._escapeMap = dart.const(dart.map({'\n': '\\n', '\r': '\\r', '\f': '\\f', '\b': '\\b', '\t': '\\t', '\v': '\\v', '': '\\x7F', [`\${${misc.expr}}`]: ''}));
  misc.main = function() {
    core.print(dart.toString(1));
    core.print(dart.toString(1.0));
    core.print(dart.toString(1.1));
    let x = 42;
    core.print(dart.equals(x, dart.wrapType(dart.dynamic)));
    core.print(dart.equals(x, dart.wrapType(misc.Generic)));
    core.print(new (misc.Generic$(core.int))().type);
    core.print(dart.equals(new misc.Derived(), new misc.Derived()));
    new (misc.Generic$(core.int))().m();
  };
  dart.fn(misc.main);
  // Exports:
  exports.misc = misc;
});

File: pkg/dev_compiler/test/codegen/misc.dart
// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.


// Codegen dependency order test
const UNINITIALIZED = const _Uninitialized();
class _Uninitialized { const _Uninitialized(); }

class Generic<T> {
  Type get type => Generic;
}

// super ==
// https://github.com/dart-lang/dev_compiler/issues/226
class Base {
  int x = 1, y = 2;
  operator==(obj) {
    return obj is Base && obj.x == x && obj.y == y;
  }
}
class Derived {
  int z = 3;
  operator==(obj) {
    return obj is Derived && obj.z == z && super == obj;
  }
}

// string escape tests
// https://github.com/dart-lang/dev_compiler/issues/227
bool _isWhitespace(String ch) =>
    ch == ' ' || ch == '\n' || ch == '\r' || ch == '\t';

const expr = 'foo';
const _escapeMap = const {
  '\n': r'\n',
  '\r': r'\r',
  '\f': r'\f',
  '\b': r'\b',
  '\t': r'\t',
  '\v': r'\v',
  '\x7F': r'\x7F', // delete
  '\${${expr}}': ''
};


main() {
  // Number literals in call expressions.
  print(1.toString());
  print(1.0.toString());
  print(1.1.toString());

  // Type literals, #184
  dynamic x = 42;
  print(x == dynamic);
  print(x == Generic);

  // Should be Generic<dynamic>
  print(new Generic<int>().type);

  print(new Derived() == new Derived()); // true
}
